packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -0,0 +1,155 @@
+      } else {
+        reject(new Error('File size greater than 250MB'));
+      }
+    });
+    output.on('error', () => {
+      reject(new Error('Failed to zip code.'));
+    });
+
+    const libPath = path.join(resourcePath, 'lib', '*');
+    const optPath = path.join(resourcePath, 'opt');
+    zip.pipe(output);
+    [...glob.sync(optPath), ...glob.sync(libPath)]
+      .filter(folder => fs.lstatSync(folder).isDirectory())
+      .forEach(folder => zip.directory(folder, path.basename(folder)));
+    zip.finalize();
+  });
+}
+
+// Check hash results for content changes, bump version if so
+async function ensureLayerVersion(context: any, layerName: string) {
+  const layerState = getLayerMetadataFactory(context)(layerName);
+  const isNewVersion = await layerState.syncVersions();
+  if (isNewVersion) {
+    const latestVersion = layerState.getLatestVersion();
+    context.print.success(`Content changes in Lambda layer ${layerName} detected. Layer version increased to ${latestVersion}`);
+    context.print.warning('Note: You need to run "amplify update function" to configure your functions with the latest layer version.');
+    await setNewVersionPermissions(context, layerName, layerState);
+  }
+  await layerState.setNewVersionHash(); // "finialize" the latest layer version
+  const storedParams = layerState.toStoredLayerParameters();
+  const additionalLayerParams = {
+    layerName,
+    build: true,
+    providerContext: {
+      provider,
+      service: ServiceName.LambdaLayer,
+      projectName: context.amplify.getProjectDetails().projectConfig.projectName,
+    },
+  };
+  const layerParameters: LayerParameters = { ...storedParams, ...additionalLayerParams };
+  updateLayerArtifacts(context, layerParameters, { cfnFile: isNewVersion });
+}
+
+async function setNewVersionPermissions(context: any, layerName: string, layerState: LayerMetadata) {
+  const defaultPermissions: PrivateLayer[] = [{ type: Permission.private }];
+  let usePrevPermissions = true;
+  const latestVersion = layerState.getLatestVersion();
+  const latestVersionState = layerState.getVersion(latestVersion);
+  const hasNonDefaultPerms =
+    latestVersionState.isPublic() || latestVersionState.listAccoutAccess().length > 0 || latestVersionState.listOrgAccess().length > 0;
+  const yesFlagSet = _.get(context, ['parameters', 'options', 'yes'], false);
+  if (yesFlagSet) {
+    context.print.warning(`Permissions from previous layer version carried forward to new version by default`);
+  } else if (hasNonDefaultPerms) {
+    usePrevPermissions = (await prompt(prevPermsQuestion(layerName))).usePrevPerms;
+  }
+  if (!usePrevPermissions) {
+    layerState.setPermissionsForVersion(latestVersion, defaultPermissions);
+  }
+}
+
+// hashes all of the layer contents as well as the files in the layer path (CFN, parameters, etc)
+export const hashLayerResource = async (layerPath: string): Promise<string> => {
+  return (await globby(['*'], { cwd: layerPath }))
```

Comment:
Can we use the standard 'glob' package and not pull in a new dependency?

Reply:
glob requires us to either use glob.sync() or promisify it ourselves, whereas globby supports promises natively. I would suggest we move toward using globby everywhere instead of glob